lib/modules/dosomething/dosomething_helpers/dosomething_helpers.share.inc

```diff
@@ -109,6 +111,11 @@ function dosomething_helpers_utm_share_paths($share_types, $path, $utm_campaign)
       )
     );
     $share_paths[$type] = $path . $utm_params;
+
+    // Append the custom share source to the URL if we have one
+    if ($custom_share_source) {
+      $share_paths[$type] .= '&source=' . $custom_share_source;
```

Comment:
Yeah I feel like it _should_ end up being funky somehow as it is but it seems like it's actually okay? The link that comes up in the tweet is something like `http://dev.dosomething.org:8888/us/campaigns/test-campaign?utm_source=dosomething&utm_medium=twitter&utm_campaign=1485_referral&source=user/5639066ba59dbfe6598b4567` which works great as far as I can tell?